stanford-earth/SE3_BLT
# SITES-8: Add TravisCI and BLT Behat integration

# READY TO MERGE

# Summary
- Changes to the `blt/project.yml` file should help BLT generate a local.yml file that includes additional paths, like `/modules` and `/profiles`, to look for behat tests.
- It also allows us to output the results as an html page, in addition to the `pretty` progress report.
- We discussed replacing `tests:behat` or creating a new, `sws:behat`. I like creating an `sws:` prefix for BLT commands we override.  That way, if we don't have to wonder, did we end up overriding this?  It's incredibly clear.
- This PR also includes a working .travis.yml file, which will run tests with the `@deploy` tag.  This is an SWS specific tag, so behat is likely to ignore any Behat tests that might be found in Drupal contributed modules under docroot/modules/contrib.  `@deploy` seems like the right tag for testing a custom site build repository because one presumes `@dev` tests were run on Stanford's custom modules as part of a CI run tied to each module repository.  So version updates in composer.json should already be vetted for deployment.  Plus, we only have ~50m within which to run tests, so `@deploy` will allow us to run a wider variety of tests, being a smaller set of our total test suite.

# Urgency
- To move forward on Caravan's phase 2, automated testing, it would be very helpful to have this merged into develop.

# Steps to Test
- Setup requires a bit of work.  Checkout this branch and `run composer update`.
- Place a feature file with an `@deploy` scenario, somewhere like: `docroot/modules/stanford/stanford_news/tests/stanford_news.feature` from https://raw.githubusercontent.com/SU-SWS/linky_clicky/5.x/sites/uat/features/high-value/stanford_news.feature.
- Run `blt sws:behat` and it should find the stanford_news.feature file.
- You should be able to see output from the tests as they run and afterwards find the full results at http://[local_url]/reports/behat/index.html.

# Associated Issues and/or People
- JIRA ticket: https://stanfordits.atlassian.net/browse/EARTH-256
- JIRA ticket: https://stanfordits.atlassian.net/browse/SITES-8
